typescript/sdk/src/token/EvmERC20WarpModule.ts

```diff
@@ -52,15 +60,105 @@ export class EvmERC20WarpModule extends HyperlaneModule<
   /**
    * Updates the Warp Route contract with the provided configuration.
    *
-   * @remark Currently only supports updating ISM or hook.
-   *
    * @param expectedConfig - The configuration for the token router to be updated.
    * @returns An array of Ethereum transactions that were executed to update the contract, or an error if the update failed.
    */
   public async update(
-    _expectedConfig: TokenRouterConfig,
+    expectedConfig: TokenRouterConfig,
+  ): Promise<AnnotatedEV5Transaction[]> {
+    TokenRouterConfigSchema.parse(expectedConfig);
+    const actualConfig = await this.read();
+
+    const updateTransactions: AnnotatedEV5Transaction[] = [];
+    updateTransactions.push(
+      ...(await this.updateIsm(actualConfig, expectedConfig)),
+    );
+
+    return updateTransactions;
+  }
+
+  /**
+   * Updates an existing Warp route ISM with a given config.
+   *
+   * @param actualConfig - The on-chain router configuration, including the ISM configuration, and address.
+   * @param expectedConfig - The expected token router configuration, including the ISM configuration.
+   * @returns Ethereum transaction that need to be executed to update the ISM configuration.
+   */
+  async updateIsm(
+    actualConfig: TokenRouterConfig,
+    expectedConfig: TokenRouterConfig,
   ): Promise<AnnotatedEV5Transaction[]> {
-    throw Error('Not implemented');
+    const updateTransactions: AnnotatedEV5Transaction[] = [];
+    const actualDeployedIsm = (
+      actualConfig.interchainSecurityModule as DerivedIsmConfig
+    ).address;
+
+    // Try to update (may also deploy) Ism with the expected config
+    const {
+      deployedIsm: expectedDeployedIsm,
+      updateTransactions: ismUpdateTransactions,
+    } = await this.deployOrUpdateIsm(actualConfig, expectedConfig);
+
+    // If an ISM is updated in-place, push the update txs
+    updateTransactions.push(...ismUpdateTransactions);
+
+    // If a new ISM is deployed, push the setInterchainSecurityModule tx
+    if (actualDeployedIsm !== expectedDeployedIsm) {
+      const contractToUpdate = MailboxClient__factory.connect(
+        this.args.addresses.deployedTokenRoute,
+        this.multiProvider.getProvider(this.args.chain),
+      );
+      updateTransactions.push({
+        annotation: `Setting ISM for Warp Route to ${expectedDeployedIsm}`,
+        chainId: Number(this.multiProvider.getChainId(this.args.chain)),
```

Comment:
I generally prefer using `getDomainId` for chainIds, just so it avoids having to cast as number and also for non-evm remotes where the chainId in the metadata is not a number

Reply:
updated.